spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfiguration.java

```diff
@@ -186,4 +192,36 @@ public RabbitMessagingTemplate rabbitMessagingTemplate(RabbitTemplate rabbitTemp
 
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(RabbitStreamTemplate.class)
+	@ConditionalOnMissingBean(RabbitStreamTemplate.class)
+	@ConditionalOnProperty(prefix = "spring.rabbitmq.listener", name = "type", havingValue = "stream")
```

Comment:
I am not sure I got that. Isn't the listener unrelated to the creation of the template? The latter is for programmatic access, isn't it?

Reply:
Agreed - using a stream producer is independent of whether listeners are also stream, by default.

Also, for consistency, this should be moved to `RabbitStreamConfiguration` (which will need some rework because the whole class is excluded via this property - move the condition from the class to the  factory `@Bean`. 

The `Environment` `@Bean` is needed for both producers and consumers.